homeassistant/components/verisure/__init__.py

```diff
@@ -83,83 +86,94 @@
 DEVICE_SERIAL_SCHEMA = vol.Schema({vol.Required(ATTR_DEVICE_SERIAL): cv.string})
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Verisure integration."""
-    global HUB  # pylint: disable=global-statement
-    HUB = VerisureHub(config[DOMAIN])
-    HUB.update_overview = Throttle(config[DOMAIN][CONF_SCAN_INTERVAL])(
-        HUB.update_overview
+    verisure = Verisure(config[DOMAIN][CONF_USERNAME], config[DOMAIN][CONF_PASSWORD])
+    coordinator = VerisureDataUpdateCoordinator(
+        hass, session=verisure, domain_config=config[DOMAIN]
     )
-    if not HUB.login():
+
+    if not await hass.async_add_executor_job(coordinator.login):
+        LOGGER.error("Login failed")
+        return False
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, lambda event: coordinator.logout()
+    )
+
+    await coordinator.async_refresh()
+    if not coordinator.last_update_success:
+        LOGGER.error("Update failed")
         return False
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, lambda event: HUB.logout())
-    HUB.update_overview()
+
+    hass.data[DOMAIN] = coordinator
 
     for platform in PLATFORMS:
-        discovery.load_platform(hass, platform, DOMAIN, {}, config)
+        await discovery.async_load_platform(hass, platform, DOMAIN, {}, config)
```

Comment:
Create a task for each platform when loading the platform to avoid a deadlock. Use `hass.async_create_task`.